### ISSUE TITLE
Viewport with empty layer ids list shows all layers

### ISSUE THREAD

**Issue by rubenthoms** [equinor/webviz-subsurface-components#2254]
I would expect a viewport with an empty layer ids list to not show anything.

**Comment by w1nklr** [comment]
It is for convenience.
If you display a single view (the usual case), you probably don't want to specify all the displayed layers on one side plus all the corresponding layer ids for the viewport...

**Comment by rubenthoms** [comment]
I understand the use case. However, in my opinion, sending in an empty list should result in showing no layers, as specified. Why can't `layerIds` be left `undefined` if one wants all layers to be displayed?

**Comment by w1nklr** [comment]
I don't know. I was not there at the initial design...
I don't really see the use case needed to display nothing. 

But if you have such an use case and if you can provide an easy way to display all the layers (and set that as DEFAULT_VIEWS), I don't object to a change.

BTW, this would be closer to Deck.gl approach: https://github.com/visgl/deck.gl/blob/4dd814805121ab70c16ea3b458593ecc92f92f1a/docs/api-reference/core/layer-manager.md?plain=1#L46

**Comment by rubenthoms** [comment]
We have a component where individual viewports can be added. To each of the latter custom layers can be added later on (in some sort of tree arrangement). When adding a layer to the first viewport, it becomes visible in all (empty) viewports although they should not have any children. We would be glad to have the possibility to temporarily have views without any layers.

**Comment by hkfb** [comment]
Agree, it makes sense to be able to show empty views, especially in the multi-viewport case.

**Comment by danielsantiago1002** [comment]
Hey @hkfb, I've worked on the issue and have a few questions before moving forward.

I found the part of the code that creates the displayed behavior:

```
    const layerFilter = useCallback(
        (args: { layer: Layer; viewport: Viewport }): boolean => {
            // display all the layers if views are not specified correctly
            if (!views?.viewports || !views?.layout) {
                return true;
            }
            const cur_view = views.viewports.find(
                ({ id }) => args.viewport.id && id === args.viewport.id
            );
            if (cur_view?.layerIds && cur_view.layerIds.length > 0) {
                const layer_ids = cur_view.layerIds;
                return layer_ids.some((layer_id) => {
                    const t = layer_id === args.layer.id;
                    return t;
                });
            } else {
                // Changing this to false to avoid displaying all layers in all views by default
                return false;
            }
        },
        [views]
    );
```

However, I found one possible Edge case that might be worth checking out:

- Currently, the user is not required to declare the "views" object. A default views object is used when one is not provided, as follows:

```
const DEFAULT_VIEWS: ViewsType = {
    layout: [1, 1],
    showLabel: false,
    marginPixels: 0,
    viewports: [{ id: "main-view", viewType: OrthographicView, layerIds: [] }],
};
```

With the proposed change to the behavior of the (layersIds) property of viewports, we would be creating a viewport without layers by default (in short, the user would be required to create the view object to obtain something other than a viewport without layers)

The question is:
- Should we add functionality to add a default view that includes all layers (only if the "views" object is undefined)? Or should we expect users to define the views object from now on? (There are breaking changes in the storybook examples.)


Thanks!

(In this issue we should also correct the examples that have viewport objects with layerIds = [ ], they should explicitly define the layer ids)

**Comment by hkfb** [comment]
> * Should we add functionality to add a default view that includes all layers (only if the "views" object is undefined)? Or should we expect users to define the views object from now on? (There are breaking changes in the storybook examples.)

That's a good observation.

We should not require `views` to be defined or change the behaviour for when `views` is undefined. I think the current behavior makes sense and is not surprising, ie. falling back to a default singular view with all layers displayed.

However, when `views` _is_ defined, we usually want to explicitly control view behavior, eg. views layout, visible layers, projection. In this case, as @rubenthoms outlines above, it's surprising that an empty `layerIds` array is interpreted as _all_ layers.

> (In this issue we should also correct the examples that have viewport objects with layerIds = [ ], they should explicitly define the layer ids)

@rubenthoms suggests above that leaving `layerIds` undefined would mean to show all layers. Could that be an option? Ie. in the stories where `layerIds = [ ]` we could just remove `layerIds` and get the same behaviour?
